### DISCUSSION TITLE
Artisan PID question. Control two damper paddles controlled via sliders.

### DISCUSSION THREAD

**Discussion by Robjones6146** [artisan-roaster-scope/artisan#1227] [General]
I am designing and building a 2 - 15kg fluid bed roaster. The roaster is fully automated using Artisan and Phidgets Vit Hub connected to many Phidgets devices that control, 2 x Belimo rotary actuators, 2 x linear actuators, 1 x solanoid valve, and 3 x relays along with 4 x thermocouples and 2 x pressure sensors. The cyclone afterburner generates the heat. The blower sucks in hot & cold air controlled by 2 damper paddles in a specific ratio. eg. 70% and 30% always equal to 100%. Cold air is assumed to be 20 C and hot 500 C. To get 100 C the hot damper will be at 16% and the cold at 84%

Some context.
I need to control two sliders, Hot air, and Cold air. They are both controlled using OUT1002 Phidgets. The (sent) voltage ranges between 0 - 10v and controls the position of the damper paddle, eg. 3 volts is 30% open and 10 volts is 100% open. The cyclone afterburner is always on, producing about 1000 - 1500 degrees C. The blower is controlled by a third slider called Loft. The loft is set depending on the batch size, eg. a 2kg roast would be set at say 60%, and a 15kg roast would be set at 90% at the start of the roast and remains unchanged.

The heat is controlled by the two damper paddles (rotating through 90 deg) that must always equal 100% for both. The cold air is at 20 C  and the hot at 500 C, therefore to get 100 C  the cold slider must be at 84% and the hot slider at 16%. (The temperatures are theoretical but serve the purpose.)

I am using a background roast profile that follows the BT. The hot air should generally run at about 60 - 100 degrees hotter than the BT and never be hotter than 275 C. I want the Artisan PID to set my Hot air slider to 30% and Cold air slider to 70% -- the BT is 64 C and the ET is 164 C (100 C greater than BT in this example) The background is the same in a perfect world.

**My question:**
How do I control two sliders using the Artisan PID Control in the correct ratio as described above?

Solutions I have considered: (but don't know how to implement - yet)
I have considered just one slider and then somehow sending 1.6v (16%) to the hot actuator when the cold actuator is at 8.4v (84%) - (10v - 8.4v = 1.6v), but don't know how to do that either. I was hoping that Artisan could control that, as I am sure it does, I just don't know how yet. Marko kindly pointed me toward symbolic formulas on a separate issue, not sure if this is relevant to what I am trying to achieve here.

I would appreciate any help, advice, and suggestion.

I am on Windows
Artisan Artisan 2.8.5 (47c8ae6)
Using Phidgets Vint Hb via USB

![Sliders](https://github.com/artisan-roaster-scope/artisan/assets/138966151/cf145040-f7cd-4241-b97e-dd32bb033767)
![Control](https://github.com/artisan-roaster-scope/artisan/assets/138966151/047392d0-c747-44f6-a4c7-ca13eb40457c)
![BelimoActuator](https://github.com/artisan-roaster-scope/artisan/assets/138966151/5462a09e-b4a6-4395-9f8b-a538ffaced1f)
![AEOOLUS15](https://github.com/artisan-roaster-scope/artisan/assets/138966151/3df3b3d5-86a2-4817-9802-8e649249e723)

**Comment by Robjones6146** [comment]
I checked the Invert Control button in the PID Control >> Target tab.
PID settings (p - 3, i - 0.05, d - 10)
Positive - Cold air
Negative - Hot air
Invert Control - checked
Duty Max - 100%
Duty Min -100%

I have some improvement, but still not working as expected. 

Can anyone assist please?

**Comment by Robjones6146** [comment]
Can anybody help me with this please?

**Reply by poundy** [reply]
I am guessing here, that no, nobody will be in a position to assist. You're making a dual control roaster from the outset. This is the kind of thing that nobody (or an extremely small number of people) will have considered, let alone done. 
Given that Artisan is a end-user-funded product, there's unlikely to be an amount of R&D (intellectual) capital the project could assist with.

**Comment by Robjones6146** [comment]
I finally figured it out. Artisan is awesome!
